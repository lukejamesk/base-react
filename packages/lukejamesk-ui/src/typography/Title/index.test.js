import React from 'react'
import { mount } from 'enzyme'
import { forEach } from 'ramda'
import { Title } from '.'

const titleTypes = ['heading1', 'heading2', 'heading3']

describe('<Title />', () => {
  it('with no type', () => {
    const wrapper = mount(
      <Title>
        My content for no title
      </Title>
    )
    expect(wrapper).toMatchSnapshot()
  })
  forEach((title) => {
    it(`should render with: ${title}`, () => {
      const wrapper = mount(
        <Title type={title}>
          My content for
          {title}
        </Title>
      )
      expect(wrapper).toMatchSnapshot()
    })
  }, titleTypes)
})
