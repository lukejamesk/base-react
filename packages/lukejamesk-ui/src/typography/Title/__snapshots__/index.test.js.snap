// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render with: heading1 1`] = `
<ForwardRef
  type="heading1"
>
  <h1>
    My content for
    heading1
  </h1>
</ForwardRef>
`;

exports[`<Title /> should render with: heading2 1`] = `
<ForwardRef
  type="heading2"
>
  <h2>
    My content for
    heading2
  </h2>
</ForwardRef>
`;

exports[`<Title /> should render with: heading3 1`] = `
<ForwardRef
  type="heading3"
>
  <h3>
    My content for
    heading3
  </h3>
</ForwardRef>
`;

exports[`<Title /> with no type 1`] = `
<ForwardRef>
  <h1>
    My content for no title
  </h1>
</ForwardRef>
`;
